### ISSUE TITLE
Reflectometry: Front Panel View

### ISSUE THREAD

**Issue by ThomasLohnert** [ISISComputingGroup/IBEX#4291]
As a reflectometry scientist, I would like to have a reflectometry perspective containing all the relevant GUI elements in one place, namely:
- scripting window
- plotting
- blocks
- dashboard 
- reflectometry front panel (this is the device screen whose name is main_screen/name of this perspective view]).

I think the right solution would be to implement this via #4290.

If we are do not have time to finish that ticket before the first IBEX reflectometer comes online, we can also consider creating a dedicated "Reflectometry" perspective with the components mentioned above.

**Comment by John-Holt-Tessella** [comment]
Solutions:

1. Add OPI into scripting window with flag in menu that adds it

**Comment by aaron-long** [comment]
Some initial findings on experimenting with adding scripting to other perspectives. The ability to add the scripting view as a part stack can be achieved for a given perspective. I added the scripting windows to the Device Screens perspective:

![stack](https://user-images.githubusercontent.com/43140680/67385304-cf140b00-f58a-11e9-84ce-fa5713c9b6d0.PNG)

This was just pointing at the `org.eclipse.ui.console.ConsoleView` ID, with the URI `bundleclass://org.eclipse.ui.workbench/org.eclipse.ui.internal.e4.compatibility.CompatibilityView`. Importantly this means that this points to the same PyDev console underneath.

**Comment by ThomasLohnert** [comment]
Work in progress - I have added a reflectometry perspective containing the scripting console and reflectometry front panel OPI. The plot window still needs some work, as I have not managed to get it to open in its intended viewpart, nor to get it to scale to it in a nice way.

I am away on leave until Jan 6. I will take my face off it for now in case someone wants to pick it up. I have committed my changes so far to ibex_gui branch `Ticket4921_refl_front_panel_view`. 

(Anyone should be able to pick this up, no reflectometry knowledge required)
